PerimeterXModule/Internals/ReverseProxy.cs

```diff
@@ -0,0 +1,240 @@
+		 * </summary>
+		 * <param name="context">HttpContext</param>
+		 */
+		public void ReversePxClient(HttpContext context)
+		{
+			Debug.WriteLine("Fetching Client", PxConstants.LOG_CATEGORY);
+			string contentType = CONTENT_TYPE_JAVASCRIPT;
+			string defaultResponse = DEFAULT_CLIENT_VALUE;
+			if (!PxConfig.FirstPartyEnabled)
+			{
+				Debug.WriteLine("First party is disabled, rendering default response", PxConstants.LOG_CATEGORY);
+				RenderPredefinedResponse(context, contentType, defaultResponse);
+			}
+			string uri = string.Format("/{0}/main.min.js", PxConfig.AppId);
+			bool success = ProcessRequest(context, PxConfig.ClientHostUrl, uri);
+
+			if (!success)
+			{
+				Debug.WriteLine("Redirect XHR returned bad status, rendering default response", PxConstants.LOG_CATEGORY);
+				RenderPredefinedResponse(context, contentType, defaultResponse);
+			}
+
+		}
+
+		/**
+		 * <summary>
+		 * Forward any sensor activities back to PerimeterX servers
+		 * </summary>
+		 * <param name="context">HttpContext</param>
+		 */
+		public void ReversePxXhr(HttpContext context)
+		{
+			string defaultResponse;
+			string contentType;
+
+			if (context.Request.Url.AbsolutePath.EndsWith(".gif"))
+			{
+				defaultResponse = Encoding.Default.GetString(DEFAULT_EMPTY_GIF_VALUE);
+				contentType = CONTENT_TYPE_GIF;
+			}
+			else
+			{
+				defaultResponse = DEFAULT_JSON_VALUE;
+				contentType = CONTENT_TYPE_JSON;
+			}
+
+			if (!PxConfig.FirstPartyEnabled || !PxConfig.FirstPartyXhrEnabled)
+			{
+				Debug.WriteLine("First party is disabled, rendering default response", PxConstants.LOG_CATEGORY);
+				RenderPredefinedResponse(context, contentType, defaultResponse);
+			}
+			string uri = context.Request.RawUrl.Replace(XhrReversePrefix, "");
+
+			string vid = null;
+			HttpCookie pxvid = context.Request.Cookies.Get("pxvid");
+			if (pxvid != null)
+			{
+				vid = pxvid.Value;
+			}
+			else if ((pxvid = context.Request.Cookies.Get("_pxvid")) != null)
+			{
+				vid = pxvid.Value;
+			}
+
+			// Clear sensitive headers
+			foreach (string sensitiveHeader in PxConfig.SensitiveHeaders)
+			{
+				if (context.Request.Headers[sensitiveHeader] != null)
+				{
+					context.Request.Headers.Remove(sensitiveHeader);
+				}
+			}
+
+			if (!string.IsNullOrEmpty(vid))
+			{
+				Debug.WriteLine(string.Format("Found VID on request, the follwoing VID will be attached to the request: {0}", vid), PxConstants.LOG_CATEGORY);
+				context.Request.Headers.Add("Cookie", string.Format("pxvid={0}", vid));
+			}
+
+			bool success = ProcessRequest(context, CollectorUrl, uri);
+			if (!success)
+			{
+				Debug.WriteLine("Redirect XHR returned bad status, rendering default response", PxConstants.LOG_CATEGORY);
+				RenderPredefinedResponse(context, contentType, defaultResponse);
+			}
+		}
+
+		/**
+		 * <summary>
+		 * Checks if this reuqest should be forwarded to PerimeterX
+		 * </summary>
+		 * <param name="uri">string, absolut uri</param>
```

Comment:
typos - request, absolute.
Also I think you can improve the summary to really describe the purpose, e.g. _Checks if this is a first party request for the Client JS sensor_.

Reply:
👍